modules/Files/endpoints.bal

```diff
@@ -0,0 +1,471 @@
+// Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/jsonutils as jsonlib;
+import ballerina/lang.array as arrays;
+import ballerina/lang.'string as stringLib;
+import ballerina/http;
+import ballerina/file;
+import ballerina/io;
+import ballerina/log;
+
+public client class AzureFileShareClient {
+    private string sasToken;
+    private string baseUrl;
+    private http:Client httpClient;
+
+    # Initalize Azure Client using the provided azureConfiguration by user
+    #
+    # + azureConfig - AzureConfiguration record
+    public function init(AzureConfiguration azureConfig) {
+        http:ClientSecureSocket? secureSocketConfig = azureConfig?.secureSocketConfig;
+        self.sasToken = stringLib:substring(azureConfig.sasToken, startIndex = 1);
+        self.baseUrl = azureConfig.baseUrl;
+        if (secureSocketConfig is http:ClientSecureSocket) {
+            self.httpClient = new (self.baseUrl, {
+                http1Settings: {chunking: http:CHUNKING_NEVER},
+                secureSocket: secureSocketConfig
+            });
+        } else {
+            self.httpClient = new (self.baseUrl, {http1Settings: {chunking: http:CHUNKING_NEVER}});
+        }
+    }
+
+    # Lists all the file shares in the  storage account
+    #
+    # + return - If success, returns ShareList record with basic details, else returns an error
+    remote function listShares(map<any> uriParameterSet = {}) returns @tainted SharesList|error {
+        string? test = check  setoptionalURIParameters(LIST_SHARES,uriParameterSet);
+        string getListPath = test is () ? (LIST_SHARE_PATH + AMPERSAND + self.sasToken) : (LIST_SHARE_PATH + test + AMPERSAND + self.sasToken);
+        http:Response response = <http:Response>check self.httpClient->get(<@untainted> getListPath);
+        if (response.statusCode == OK) {
+            xml formattedXML = check xmlFormatter(check response.getXmlPayload()/<Shares>);
+            json jsonValue = check jsonlib:fromXML(formattedXML);
+            return <SharesList>check jsonValue.cloneWithType(SharesList);
+        } else {
```

Comment:
else part can be removed and reduce the no. of lines since it make sense to reach the error only if statusCode != OK. So just can return the error in one line

Reply:
We check the =OK and gets the payload after converting to a record type. Even we remove else, we have to add that converting part.